packages/visual-dev/src/components/Buttons.tsx

```diff
@@ -0,0 +1,352 @@
+    }
+  }
+
+  @keyframes dash {
+    0% {
+      stroke-dashoffset: 1000;
+    }
+    100% {
+      stroke-dashoffset: 0;
+    }
+  }
+
+  @-webkit-keyframes dash-check {
+    0% {
+      stroke-dashoffset: -100;
+    }
+    100% {
+      stroke-dashoffset: 900;
+    }
+  }
+
+  @keyframes dash-check {
+    0% {
+      stroke-dashoffset: -100;
+    }
+    100% {
+      stroke-dashoffset: 900;
+    }
+  }
+`;
+
+const rotate = keyframes`
+   from {      
+     transform: rotate(0deg);    
+    }    
+    to {      
+      transform: rotate(360deg);    
+    }
+`;
+
+export const RingDefault = styled.div`
+  display: inline-block;
+
+  div {
+    box-sizing: border-box;
+    display: block;
+    position: absolute;
+    border-radius: 50%;
+    animation: ${rotate} 1.7s cubic-bezier(0.5, 0, 0.5, 1) infinite;
+  }
+
+  div:nth-child(1) {
+    animation-delay: -0.45s;
+  }
+
+  div:nth-child(2) {
+    animation-delay: -0.3s;
+  }
+
+  div:nth-child(3) {
+    animation-delay: -0.15s;
+  }
+`;
+
+export const ButtonSpinnerStyle = styled(RingDefault)`
+  position: relative;
+  bottom: 12px;
+  padding: 0px 10px;
+  div {
+    width: 15px;
+    height: 15px;
+    border: 2px solid #f5a623;
+    border-color: #f5a623 transparent transparent transparent;
+  }
+`;
+
+export const ButtonSpinner = () => {
+  return (
+    <ButtonSpinnerStyle>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+    </ButtonSpinnerStyle>
+  );
+};
+
+export const PrimaryButton = ({
+  children,
+  onClick,
+  padding,
+  buttonColor,
+  darkercolor,
+  disabledcolor,
+  loading,
+  ...props
+}: PBProps) => {
+  return (
+    <PrimaryButtonStyled
+      {...props}
+      padding={padding}
+      buttonColor={buttonColor}
+      darkercolor={darkercolor}
+      disabledcolor={disabledcolor}
+      onClick={onClick}
+      disabled={loading || props.disabled}
+    >
+      {children}
+      {loading && <ButtonSpinner />}
+    </PrimaryButtonStyled>
+  );
+};
+
+// TODO: make one Primary Button to rule them all!
+const PrimaryButtonStyled = styled.button<PBProps>`
+  border-radius: 20px;
+  min-width: 100px;
+  padding: ${(props) => props.padding || `3px 19px`};
+  background: ${(props) => props.buttonColor || `#f5a841`};
+  border: 1px solid ${(props) => props.buttonColor || `#f5a841`};
+  color: #fff;
+  font-weight: 600;
+  font-size: 13px;
+  font-family: "Helvetica Neue", Helvetica, sans-serif;
+  &:hover,
+  &:active,
+  &:focus {
+    background: ${(props) => props.darkercolor || `#DC8F28`};
+    border: 1px solid ${(props) => props.buttonColor || `#DC8F28`};
+    color: #fff;
+    text-decoration: none;
+    outline: none;
+  }
+  :disabled {
+    background: ${(props) => props.disabledcolor || `#E2E2E2`};
+    border: 1px solid ${(props) => props.disabledcolor || `#E2E2E2`};
+    pointer-events: none;
+  }
+  ${ButtonDiv} & {
+    margin: 0 8px;
+  }
+`;
+
+
+
+// Primary Button ends
+
+interface IPrimaryButton extends PBProps {
+  onClick: any;
```

Comment:
This could be typed properly, from the usage it looks like this takes a SyntheticEvent and is expected to return a Promise<void>, so it could be:
```suggestion
  onClick: (e: SyntheticEvent) -> Promise<void>;
```

Reply:
Implemented with a slight amendment `(e: SyntheticEvent) => Promise<void> | void`. This maintains compatibility with the often used in testing `onClick={() => console.log("submitted!")}`. Not totally sure if that's a good idea or not.

Reply:
I would keep the signature stable, and in your tests return `Promise.resolve(console.log("submitted!")`